src/main/java/com/rackspace/ceres/app/services/MetricDeletionService.java

```diff
@@ -79,12 +81,21 @@ public MetricDeletionService(DataTablesStatements dataTablesStatements,
    */
   private Mono<Empty> deleteMetricsByTenantId(String tenant, Instant start, Instant end) {
     log.debug("Deleting metrics for tenant: {}", tenant);
-    Flux<String> seriesSetHashes = metricDeletionHelper
-        .getSeriesSetHashFromRawOrDownsampled(tenant, start, end);
-    return deleteMetadataByTenantId(tenant, seriesSetHashes)
-        .then(deleteMetricsFromDownsampled(tenant, start, end))
-        .then(deleteMetricsFromRaw(tenant, start, end))
-        .then(Mono.empty());
+    Flux<String> seriesSetHashes = null;
+    if (start != null) {
+      return deleteMetricsFromDownsampled(tenant, start, end)
+          .then(deleteMetricsFromRaw(tenant, start, end))
+          .then(Mono.empty());
+    } else {
+      start = Instant.now().minus(getHighestTTLGranularity().getTtl());
```

Comment:
if start (and end) are not defined, shouldn't you delete all the data for this tenant id? Why are you setting a start value here? Also where are the metadata related to this tenant deleted?

Reply:
In order to delete all the metrics data along with its related metadata for that tenant, we need timeSlot and seriesSetHashes. So, at first I am querying all the seriesSetHashes to delete metadata in https://github.com/racker/ceres/blob/42d90969cbb4a8d45384acbe46d1c7c850e0911b/src/main/java/com/rackspace/ceres/app/services/MetricDeletionService.java#L94 and then deleting its metrics data afterwards.

Reply:
`delete from <table> where tenant = 't-1';` 

Why not do like this?